### PR TITLE
dnsproxy: 0.54.0 -> 0.65.0

### PR DESCRIPTION
**Pull request by csutora**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/dnsproxy/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "dnsproxy";
-  version = "0.54.0";
+  version = "0.65.0";
 
   src = fetchFromGitHub {
     owner = "AdguardTeam";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-wA88v8zF5sq3NQwKh8g4k/COviuaegGn6bBTzBMcdGM=";
+    sha256 = "sha256-TE33gwEfOLWa32lHM7gWsrtUsVX0B5pWtxMfyD36AyQ=";
```
Comment (dotlambda) [inline pkgs/tools/networking/dnsproxy/default.nix:11]:
```suggestion
    hash = "sha256-TE33gwEfOLWa32lHM7gWsrtUsVX0B5pWtxMfyD36AyQ=";
```

File: pkgs/tools/networking/dnsproxy/default.nix [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 buildGoModule rec {
   pname = "dnsproxy";
-  version = "0.54.0";
+  version = "0.65.0";
 
   src = fetchFromGitHub {
     owner = "AdguardTeam";
     repo = pname;
```
Comment (dotlambda) [inline pkgs/tools/networking/dnsproxy/default.nix:9]:
```suggestion
    repo = "dnsproxy";
```
#277994

**Comment by contrun** [comment]
Thank you for the contribution. Unfortunately, I can't build this with `nixpkgs-review pr 288686`. The error is

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/aq86n4vd0ihdgilvpq0fcvls80za08nx-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Building subPackage .
go: go.mod requires go >= 1.21.7 (running go 1.21.6; GOTOOLCHAIN=local)
```

**Review by IogaMaster — CHANGES_REQUESTED** [review]

**Comment by IogaMaster** [comment]
Also, reproducing the issue @contrun is facing.

File: pkgs/tools/networking/dnsproxy/default.nix [hunk 3]
```diff
@@ -2,13 +2,13 @@
 
 buildGoModule rec {
   pname = "dnsproxy";
-  version = "0.54.0";
+  version = "0.65.0";
 
   src = fetchFromGitHub {
     owner = "AdguardTeam";
-    repo = pname;
+    repo = "dnsproxy";
```
[the remark this answers is not in the document]
Comment (csumpasd) [reply pkgs/tools/networking/dnsproxy/default.nix:9]: @dotlambda suggested the edit, I thought I was making a mistake.

[the remark this answers is not in the document]
Comment (IogaMaster) [reply pkgs/tools/networking/dnsproxy/default.nix:9]: I see, I was unaware of the new movement away from `repo= pname`

**Review by IogaMaster — APPROVED** [review]

**Comment by csumpasd** [comment]
If I understand correctly, the version of go is the problem, and #287998 seems to be working, so I'll close this pr.

I'm really sorry for not testing the build, I honestly couldn't figure out how to do so, but guessing from your responses it's done with `nixpkgs-review pr`.

Thanks everyone!

**Comment by IogaMaster** [comment]
Okay! Thanks for your effort! :smile:

**Comment by diogotcorreia** [comment]
@csumpasd You can also just build the package using `nix-build -A dnsproxy` (or with the new CLI, `nix build .#dnsproxy`).

Hope to see you contribute again :)

**Comment by csumpasd** [comment]
@diogotcorreia Thanks! Does that build the package from the local nixpkgs if I'm in the folder? Or is that a different command?

**Comment by diogotcorreia** [comment]
@csumpasd Yes, that's exactly what it does!

**Comment by csumpasd** [comment]
@diogotcorreia Thanks! Hope to contribute again, properly next time!

**Pull request closed without merging**